server/internal/router/pod.go

```diff
@@ -0,0 +1,75 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/llm-operator/inference-manager/server/internal/k8s"
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// podEventHandler is the event handler for managing pod.
+type podEventHandler struct {
+	m     *routeMap
+	errCh chan error
+}
+
+var _ k8s.EventHandlerWithContext = &podEventHandler{}
+
+// newPodEventHandler creates a new PodEventHandler.
+func newPodEventHandler(
+	m *routeMap,
+	errCh chan error,
+) *podEventHandler {
+	return &podEventHandler{
+		m:     m,
+		errCh: errCh,
+	}
+}
+
+// ProcessAdd processed the add event.
+func (h *podEventHandler) ProcessAdd(ctx context.Context, obj interface{}) {
+	// o, ok := obj.(runtime.Object)
+	pod, ok := obj.(*apiv1.Pod)
+	if !ok {
+		h.errCh <- fmt.Errorf("cast runtime.Object type")
+		return
+	}
+	log.Printf("Create a pod, ip is %s\n", pod.Status.PodIP)
+	if pod.Status.PodIP != "" {
+		h.m.addServer(pod.Status.PodIP)
+	}
+}
+
+// ProcessUpdate processes the update event.
+func (h *podEventHandler) ProcessUpdate(ctx context.Context, oldObj, newObj interface{}) {
+	newPod, ok := newObj.(*apiv1.Pod)
+	if !ok {
+		h.errCh <- fmt.Errorf("cast runtime.Object type")
+		return
+	}
+	oldPod, ok := oldObj.(*apiv1.Pod)
+	if !ok {
+		h.errCh <- fmt.Errorf("cast runtime.Object type")
+		return
+	}
+	if newPod.Status.PodIP == oldPod.Status.PodIP {
+		return
+	}
+	log.Printf("Update a pod, new IP is %s, old IP is %s\n", newPod.Status.PodIP, oldPod.Status.PodIP)
+	if oldPod.Status.PodIP == "" {
```

Comment:
The code here is correct, but it implicitly assumes that a pod IP only changes from an empty value to a non-empty value once. It looks a bit weird that we won't do anything here if a pod IP changes from a non-empty value to a different non-empty value here.

How about changing the code to be more explicit?

```go
if oldPod.Status.PodIP != "" {
  h.errCh <- fmt.Errorf("unexpected ip change)
  return
}
h.m.addServer(newPod.Status.PodIP)
```